src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -100,14 +96,9 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     sagaEntity.ConcurrencyCheck(sagaData, version);
                 }
 
-                data.AddOrUpdate(sagaData.Id, id => new VersionedSagaEntity
-                {
-                    SagaData = DeepClone(sagaData)
-                }, (id, original) => new VersionedSagaEntity
-                {
-                    SagaData = DeepClone(sagaData),
-                    VersionCache = original.VersionCache
-                });
+                data.AddOrUpdate(sagaData.Id,
+                    id => new VersionedSagaEntity(DeepClone(sagaData)),
+                    (id, original) => new VersionedSagaEntity(DeepClone(sagaData), original));
```

Comment:
@timbussmann have I look. I inlined the DeepClone. Caveat the serializer is now static since we don't want a serializer per `VersionedSagaEntity` when good I squash and push again